mrp_mto_with_stock/models/mrp_production.py

```diff
@@ -30,74 +29,69 @@ def action_assign(self):
             warehouse = production.location_src_id.get_warehouse()
             mto_with_no_move_dest_id = warehouse.mrp_mto_mts_forecast_qty
             for move in self.move_raw_ids:
+                group = new_move = procurement = False
+                qty_to_procure = 0.0
                 if (move.state == 'confirmed' and move.location_id in
                         move.product_id.mrp_mts_mto_location_ids and not
                         mto_with_no_move_dest_id):
-                    domain = [('product_id', '=', move.product_id.id),
-                              ('move_dest_id', '=', move.id)]
                     if move.group_id:
-                        domain.append(('group_id', '=', move.group_id.id))
-                    procurement = self.env['procurement.order'].search(domain)
+                        procurement = True
```

Comment:
Why are you asuming that if you have group you don't have to procure?

Reply:
@lreficent will improve these. Appreciate your feedback.